translator/trace/zipkin/zipkinv2_to_traces.go

```diff
@@ -0,0 +1,454 @@
+	traceData := pdata.NewTraces()
+	if len(zipkinSpans) == 0 {
+		return traceData, nil
+	}
+
+	sort.Sort(byOTLPTypes(zipkinSpans))
+
+	rss := traceData.ResourceSpans()
+	prevServiceName := ""
+	prevInstrLibName := ""
+	rsCount := rss.Len()
+	ilsCount := 0
+	spanCount := 0
+	var curRscSpans pdata.ResourceSpans
+	var curILSpans pdata.InstrumentationLibrarySpans
+	var curSpans pdata.SpanSlice
+	for _, zspan := range zipkinSpans {
+		if zspan == nil {
+			continue
+		}
+		tags := copySpanTags(zspan.Tags)
+		localServiceName := extractLocalServiceName(zspan)
+		if localServiceName != prevServiceName {
+			prevServiceName = localServiceName
+			rss.Resize(rsCount + 1)
+			curRscSpans = rss.At(rsCount)
+			curRscSpans.InitEmpty()
+			rsCount++
+			populateResourceFromZipkinSpan(tags, localServiceName, curRscSpans.Resource())
+			prevInstrLibName = ""
+			ilsCount = 0
+		}
+		instrLibName := extractInstrumentationLibrary(zspan)
+		if instrLibName != prevInstrLibName || ilsCount == 0 {
+			prevInstrLibName = instrLibName
+			curRscSpans.InstrumentationLibrarySpans().Resize(ilsCount + 1)
+			curILSpans = curRscSpans.InstrumentationLibrarySpans().At(ilsCount)
+			curILSpans.InitEmpty()
+			ilsCount++
+			populateILFromZipkinSpan(tags, instrLibName, curILSpans.InstrumentationLibrary())
+			spanCount = 0
+			curSpans = curILSpans.Spans()
+		}
+		curSpans.Resize(spanCount + 1)
+		err := zSpanToInternal(zspan, tags, curSpans.At(spanCount))
+		if err != nil {
+			return traceData, err
+		}
+		spanCount++
+	}
+
+	return traceData, nil
+}
+
+func zSpanToInternal(zspan *zipkinmodel.SpanModel, tags map[string]string, dest pdata.Span) error {
+	dest.InitEmpty()
+
+	dest.SetTraceID(pdata.TraceID(tracetranslator.UInt64ToByteTraceID(zspan.TraceID.High, zspan.TraceID.Low)))
+	dest.SetSpanID(pdata.SpanID(tracetranslator.UInt64ToByteSpanID(uint64(zspan.ID))))
+	if value, ok := tags[tracetranslator.TagW3CTraceState]; ok {
+		dest.SetTraceState(pdata.TraceState(value))
+		delete(tags, tracetranslator.TagW3CTraceState)
+	}
+	parentID := zspan.ParentID
+	if parentID != nil && *parentID != zspan.ID {
+		dest.SetParentSpanID(pdata.SpanID(tracetranslator.UInt64ToByteSpanID(uint64(*parentID))))
+	}
+
+	dest.SetName(zspan.Name)
+	startNano := zspan.Timestamp.UnixNano()
+	dest.SetStartTime(pdata.TimestampUnixNano(startNano))
+	if zspan.Duration.Nanoseconds() > 0 {
+		dest.SetEndTime(pdata.TimestampUnixNano(startNano + zspan.Duration.Nanoseconds()))
+	}
+	dest.SetKind(zipkinKindToSpanKind(zspan.Kind, tags))
+
+	populateSpanStatus(tags, dest.Status())
+	if err := zTagsToSpanLinks(tags, dest.Links()); err != nil {
+		return err
+	}
+
+	attrs := dest.Attributes()
+	attrs.InitEmptyWithCapacity(len(tags))
+	if err := zTagsToInternalAttrs(zspan, tags, attrs); err != nil {
+		return err
+	}
+
+	err := populateSpanEvents(zspan, dest.Events())
+	return err
+}
+
+func populateSpanStatus(tags map[string]string, status pdata.SpanStatus) {
```

Comment:
The previous implementation used [statusMapper](https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/trace/zipkin/status_code.go#L51) to infer the status from additional tags like `http.status_code` https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/trace/zipkin/zipkinv2_to_protospan.go#L239-L252

Reply:
Status is going to be removed from the specification. We should wait until that happens to revise this.